### PR TITLE
VMs lifecycle tests: fix assumption for compute nodes

### PR DESCRIPTION
**Pull request by lukas-bednar**
**What this PR does / why we need it**:

Some test requires to work with compute node - these are counting with
fact that all nodes are compute, which is wrong assumption.

Signed-off-by: Lukas Bednar <lbednar@redhat.com>

```release-note
NONE
```

### CODE REVIEW

**Comment by lukas-bednar** [comment]
Tests, which were affected by this PR, passed in CI.
```
------------------------------
â€¢ [SLOW TEST:41.762 seconds]
VMIlifecycle
/root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:71
  Creating a VirtualMachineInstance
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:93
    with non default namespace
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:598
      should log libvirt start and stop lifecycle events of the domain
      /root/go/src/kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
        kubevirt-test-default
        /root/go/src/kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
------------------------------
â€¢ [SLOW TEST:34.459 seconds]
VMIlifecycle
/root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:71
  Creating a VirtualMachineInstance
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:93
    with non default namespace
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:598
      should log libvirt start and stop lifecycle events of the domain
      /root/go/src/kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
        kubevirt-test-alternative
        /root/go/src/kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
------------------------------
â€¢ [SLOW TEST:19.966 seconds]
VMIlifecycle
/root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:71
  Delete a VirtualMachineInstance
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:932
    with grace period greater than 0
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:1009
      should run graceful shutdown
      /root/go/src/kubevirt.io/kubevirt/tests/vmi_lifecycle_test.go:1010
------------------------------
```

**Comment by rmohr** [comment]
@lukas-bednar they fail at least on 1.10.4 and 1.11.0: https://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/2049//artifact/check-patch.k8s-1.10.4-release.el7.x86_64/mock_logs/script/stdout_stderr.log

 Did not check others. In these clusters only master have this label.

**Comment by lukas-bednar** [comment]
>@lukas-bednar they fail at least on 1.10.4 and 1.11.0: https://jenkins.ovirt.org/job/kubevirt_kubevirt_standard-check-pr/2049//artifact/check-patch.k8s-1.10.4-release.el7.x86_64/mock_logs/script/stdout_stderr.log
>
> Did not check others. In these clusters only master have this label.

Aha, I have checked only OCP logs - my bad ...

File: tests/vmi_lifecycle_test.go [hunk 1]
```diff
@@ -598,9 +598,9 @@ var _ = Describe("VMIlifecycle", func() {
 		Context("with non default namespace", func() {
 			table.DescribeTable("should log libvirt start and stop lifecycle events of the domain", func(namespace string) {
 
-				nodes, err := virtClient.CoreV1().Nodes().List(metav1.ListOptions{})
-				Expect(err).ToNot(HaveOccurred(), "Should list nodes")
-				Expect(nodes.Items).ToNot(BeEmpty(), "There should be some node")
+				nodes, err := virtClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/compute=true"})
```
[the remark this answers is not in the document]
Comment (lukas-bednar) [reply tests/vmi_lifecycle_test.go]: OK I will change it to `kubevirt.io/schedulable=true`

**Comment by lukas-bednar** [comment]
@rmohr I believe that it looks better now ...

File: tests/vmi_lifecycle_test.go [hunk 2]
```diff
@@ -598,9 +598,9 @@ var _ = Describe("VMIlifecycle", func() {
 		Context("with non default namespace", func() {
 			table.DescribeTable("should log libvirt start and stop lifecycle events of the domain", func(namespace string) {
 
-				nodes, err := virtClient.CoreV1().Nodes().List(metav1.ListOptions{})
-				Expect(err).ToNot(HaveOccurred(), "Should list nodes")
-				Expect(nodes.Items).ToNot(BeEmpty(), "There should be some node")
+				nodes, err := virtClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "kubevirt.io/schedulable=true"})
```
Comment (cynepco3hahue) [inline tests/vmi_lifecycle_test.go]: Can you move it to a separate method? It can be pretty useful.

Comment (lukas-bednar) [reply tests/vmi_lifecycle_test.go]: Done ;-)

**Pull request merged**